app/extensions/edm/__init__.py

```diff
@@ -313,7 +312,7 @@ def _request(
         verbose=True
     ):
         if ensure_initialized:
```

Comment:
This is a strange check. 
Nothing calls it with ensure_initialized to false anyway, so my first thought is ditch the parameter and always call the function.
Second thought is that as _request is a private function., the ensure stuff should be called on entry points into the class, not on private methods and this should be removed entirely

Reply:
not sure of the initial intention of this, other than guessing that it just left open the _option_ of calling `ensure_initialzed=False` when you wanted to skip the step?